packages/react-integration/demo-app-ts/src/components/demos/DualListSelector/DualListSelectorDemo.tsx

```diff
@@ -0,0 +1,173 @@
+            availableDescending: !prevState.availableDescending
+          };
+        });
+      }
+
+      if (panel === 'chosen') {
+        this.setState(prevState => {
+          const chosen = prevState.chosenOptions.sort((a: any, b: any) => {
+            let returnValue = 0;
+            if (a.props.children > b.props.children) {
+              returnValue = -1;
+            }
+            if (a.props.children < b.props.children) {
+              returnValue = 1;
+            }
+            if (prevState.chosenDescending) {
+              returnValue = returnValue * -1;
+            }
+            return returnValue;
+          });
+          return {
+            chosenOptions: chosen,
+            chosenDescending: !prevState.chosenDescending
+          };
+        });
+      }
+    };
+
+    this.onListChange = (newAvailableOptions, newChosenOptions) => {
+      this.setState({
+        availableOptions: newAvailableOptions,
+        chosenOptions: newChosenOptions
+      });
+    };
+
+    this.onToggle = (isOpen, pane) => {
+      this.setState((prevState: DualListSelectorState) => ({
+        isAvailableKebabOpen: pane === 'available' ? isOpen : prevState.isAvailableKebabOpen,
+        isChosenKebabOpen: pane === 'chosen' ? isOpen : prevState.isChosenKebabOpen
+      }));
+    };
+
+    this.filterOption = (option: any, input: string) => option.props.children.includes(input);
+  }
+
+  render() {
+    const dropdownItems = [
+      <DropdownItem key="link">Link</DropdownItem>,
+      <DropdownItem key="action" component="button">
+        Action
+      </DropdownItem>,
+      <DropdownItem key="second action" component="button">
+        Second Action
+      </DropdownItem>
+    ];
+
+    const availableOptionsActions = [
+      <Button
+        variant={ButtonVariant.plain}
+        onClick={() => this.onSort('available')}
+        aria-label="Sort"
+        key="availableSortButton"
+      >
+        {this.state.availableDescending ? <PficonSortCommonDescIcon /> : <PficonSortCommonAscIcon />}
+      </Button>,
+      <Dropdown
+        toggle={<KebabToggle onToggle={isOpen => this.onToggle(isOpen, 'available')} id="toggle-id-6" />}
+        isOpen={this.state.isAvailableKebabOpen}
+        isPlain
+        dropdownItems={dropdownItems}
+        key="availableDropdown"
+      />
+    ];
+
+    const chosenOptionsActions = [
+      <Button
+        variant={ButtonVariant.plain}
+        onClick={() => this.onSort('chosen')}
+        aria-label="Sort"
+        key="chosenSortButton"
+      >
+        {this.state.chosenDescending ? <PficonSortCommonDescIcon /> : <PficonSortCommonAscIcon />}
+      </Button>,
+      <Dropdown
+        toggle={<KebabToggle onToggle={isOpen => this.onToggle(isOpen, 'chosen')} id="toggle-id-6" />}
+        isOpen={this.state.isChosenKebabOpen}
+        isPlain
+        dropdownItems={dropdownItems}
+        key="chosenDropdown"
+      />
+    ];
+
+    return (
+      <DualListSelector
```

Comment:
Can we add example to make sure we use all the props the verify typing and expected behaviour?